crates/runtime/src/http/auth.rs

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use axum::{
+    body::Body,
+    http::StatusCode,
+    http::{Request, Response},
+};
+use futures::future::BoxFuture;
+use runtime_auth::{AuthVerdict, HttpAuth};
+use std::sync::Arc;
+use std::task::{Context, Poll};
+use tower::{Layer, Service};
+
+#[derive(Clone)]
+pub struct AuthLayer {
+    auth_verifier: Arc<dyn HttpAuth + Send + Sync>,
+}
+
+impl AuthLayer {
+    pub fn new(auth_verifier: Arc<dyn HttpAuth + Send + Sync>) -> Self {
+        Self { auth_verifier }
+    }
+}
+
+impl<S> Layer<S> for AuthLayer {
```

Comment:
This is a tower Layer, which you can think of as a factory for building tower Services. In this case, we are just wrapping the given tower Service with our `AuthMiddleware` Service implementation